packages/interactivity-router/src/index.ts

```diff
@@ -107,15 +115,25 @@ const regionsToVdom: RegionsToVdom = async ( dom, { vdom } = {} ) => {
 	}
 	const title = dom.querySelector( 'title' )?.innerText;
 	const initialData = parseServerData( dom );
-	return { regions, head, title, initialData };
+	return { regions, styles, scriptModules, title, initialData, url };
 };
 
 // Render all interactive regions contained in the given page.
 const renderRegions = async ( page: Page ) => {
+	// Wait for styles and modules to be ready.
+	await Promise.all( [
+		...page.styles,
+		...page.scriptModules.map(
+			( src ) => import( /* webpackIgnore: true */ src )
+		),
+	] );
+	// Replace style sheets.
+	const styles = await Promise.all( page.styles );
```

Comment:
Simply to understand it: All these style promises are already resolved in the second `await`, and the only thing you are doing is collecting them in an array, is that correct?

Reply:
That's exactly it. I guess it could be changed to something like this:

```suggestion
	// Wait for styles and modules to be ready.
	const [ styles ] = await Promise.all( [
		Promise.all( page.styles ),
		Promise.all(
			page.scriptModules.map(
				( src ) => import( /* webpackIgnore: true */ src )
			)
		),
	] );
	// Replace style sheets.
```

Reply:
Yes, good catch. 

We need to rethink when `actions.prefetch` should be resolved, certainly after the HTML and CSS have been loaded. What I'm not so sure about is how the preloading of the scripts should be managed.

Reply:
Let's move this conversation to your new PR.

- https://github.com/woocommerce/gutenberg/pull/10